ios/Classes/components/applePay/ApplePayComponentManager.swift

```diff
@@ -0,0 +1,34 @@
+@_spi(AdyenInternal)
+import Adyen
+
+class ApplePayComponentManager {
+    private let sessionHolder: SessionHolder
+    var applePayComponent: BaseApplePayComponent?
+
+    init(sessionHolder: SessionHolder) {
+        self.sessionHolder = sessionHolder
+    }
+ 
+    func isApplePayAvailable(instantPaymentComponentConfigurationDTO: InstantPaymentConfigurationDTO, callback: (Result<InstantPaymentSetupResultDTO, Error>) -> Void) {
+        guard let applePayConfiguration = try? instantPaymentComponentConfigurationDTO.mapToApplePayConfiguration(),
+              let adyenContext = try? instantPaymentComponentConfigurationDTO.createAdyenContext()
+        else {
+            callback(Result.failure(PlatformError(errorDescription: "Apple Pay configuration error occurred")))
+            return
+        }
+            
+        if sessionHolder.session != nil, sessionHolder.sessionDelegate != nil {
+            applePayComponent = ApplePaySessionComponent(
```

Comment:
Feels like a hidden sideeffect to setup a component on a `isApplePayAvailable` call

Reply:
Maybe rename the function to `setupIfAvailable` or something